crates/types/src/transaction/layer.rs

```diff
@@ -2,3 +2,9 @@
 #[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
 #[repr(transparent)]
 pub struct Layer(pub u64);
+
+impl Default for Layer {
+    fn default() -> Self {
+        Self(0)
+    }
+}
```

Comment:
Can you please explain to me why 0 is the `Default` `Layer`? I think I'm missing something

Reply:
BTW we might want to turn this into a `type` alias in the future.

Reply:
> Can you please explain to me why 0 is the `Default` `Layer`? I think I'm missing something

Couldn't think of any other value to use, when we don't really care about the `Layer`.
We could maybe use `Option<..>` but it seems over-kill in this case IMHO.

Reply:
> BTW we might want to turn this into a `type` alias in the future.

Probably we could. However,  sometimes we might end up wanting to implement a trait or two, and we won't be able to do
that when using an `alias` type.

Reply:
I agree on the `type` vs `struct` issue after some thinking.

Reply:
I don't think `Layer` should have a `Default`. Semantically, it doesn't make much sense. Considering that it's only used for tests, how about we add `#[cfg(test)]` to `Context::with_state` and fill it with dummy data (0 is fine, just like you said)?

```rust

    /// Creates a new [`Context`] with the given `state` parameter.
    ///
    /// Sets dummy values for all remaining fields.
    ///
    /// # Notes
    ///
    /// This function/method is only available during testing.
    #[cfg(test]
    // (We can also move this to `runtime_tests.rs`)
    pub fn with_state(state: State) -> Self {
        Self {
            tx_id: TransactionId::zeros(),
            layer: Layer::default(),
            state: state,
        }
    }
```